### PR TITLE
BT-614 Remove Alibaba/Aliyun backend

### PR DESCRIPTION
**Pull request by kraefrei**

### CODE REVIEW

File: CHANGELOG.md
Comment (aednichols) [inline CHANGELOG.md:32]: Cromwell 80 already released [0] so this change will be in the next version.

You might get into an awkward state where Janet releases 81 before this PR merges, in which case it would become 82. Annoying, but rare and relatively easy to fix.

[0] https://github.com/broadinstitute/cromwell/releases

**Review by aednichols — APPROVED** [review]
Looking good!

Doing a full-text search of the project in IntelliJ, looks like there's still one reference to clean up in `CallCaching.md`

**Comment by jgainerdewar** [comment]
Good catch Adam!
I also see a reference to `CROMWELL_BUILD_BCS_CLUSTER_ID` in `src/ci/docker-compose-horicromtal.yml`

File: CHANGELOG.md
Comment (kraefrei) [reply CHANGELOG.md]: In chat, talked with Janet and it sounded like we'd try to put this in the 81 release as well.

**Review by jgainerdewar — APPROVED** [review]
LGTM pending tests passing again!

**Review by aednichols — APPROVED** [review]

File: docs/filesystems/Filesystems.md
Comment (mcovarr) [inline docs/filesystems/Filesystems.md:56]: Pedantic nit
```suggestion
- **S3 filesystem is experimental.** 
```

File: CHANGELOG.md
Comment (mcovarr) [inline CHANGELOG.md:11]: This should maybe say 'Alibaba' somewhere? IMHO it sounds like a broader change than it actually is.

Comment (aednichols) [reply CHANGELOG.md:11]: +100

**Review by aednichols — APPROVED** [review]
Thank you for the completeness and attention to detail! 👍

**Comment by aednichols** [comment]
I believe it is implicitly imported as a transitive dependency, which is to say we don't have any actual import statements for it... it will just disappear automagically with Alibaba.

**Pull request merged**
